src/test/kotlin/cacophony/codegen/registers/SpillHandlingTest.kt

```diff
@@ -476,9 +508,106 @@ class SpillHandlingTest {
                 instructionCovering,
                 functionHandler,
                 listOf(block),
+                mockLiveness,
                 registerAllocation,
                 setOf(spareReg),
+                mockGraphColoring,
             )
         }
     }
+
+    @Test
+    fun `allocates the same frame memory to spills colored with the same color`() {
```

Comment:
This test is already long in terms of lines of code, but describes a quite simple scenario - maybe add e.g. a third register interfering with the other two?

Reply:
added one reg that interferes with the two, and also one non-spilled reg to check if graph coloring is called with subgraph of interference graph induced by only spilled registers